### ISSUE TITLE
More Easily Set Default Configuration Directory at Build Time

### ISSUE THREAD

**Issue by FOSSilizedDaemon** [atlas-engineer/nyxt#2582]
**Is your feature request related to a problem? Please describe.**
I want to store my default `nyxt` configuration directory in `~/.library/generic/common-lisp/nyxt` instead of `~/.config/nyxt`, but in order to achieve this properly I have to edit the actual source-code prior to building (otherwise I still need a file in `XDG_CONFIG_HOME/nyxt` to tell `nyxt` where to look for the configuration directory).

**Describe the solution you'd like**
Could you add a single file or config.mk to set the directory to use as the default config directory?

**Describe alternatives you've considered**
No other alternatives I know of.

**Additional context**
I am weird and am trying to move all executed or actual scripting/programming configurations (lisp, lua, shell-script, etc) to `~/.library`, got the idea from Plan9, so I can set `XDG_CONFIG_HOME` to only r+w instead of r+w+e.

**Comment by Ambrevar** [comment]
Wouldn't it be better to make this customizable at runtime via an environment variable, say `NYXT_CONFIG_FILE`?

**Comment by FOSSilizedDaemon** [comment]
> Wouldn't it be better to make this customizable at runtime via an environment variable, say `NYXT_CONFIG_FILE`?

100%. I guess to put it better, could we add *an option*, either compile time or run time, to more easily set the nyxt configuration directory.

**Comment by Ambrevar** [comment]
OK, this is an easy one then.

**Comment by FOSSilizedDaemon** [comment]
> OK, this is an easy one then.

Awesome! Thank you!

**Comment by aadcg** [comment]
I don't think we should support this. As mentioned by the OP, the right way to patch the source files.
